src/space.js

```diff
@@ -37,8 +44,12 @@ class Space {
       const spaceAddress = await this._store._load()
 
       const entries = await this._rootStore.iterator({ limit: -1 }).collect()
-      if (!entries.find(entry => entry.payload.value.odbAddress.indexOf(nameToSpaceName(this._name)) !== -1)) {
-        this._rootStore.add({ odbAddress: spaceAddress })
+      const entry = entries.find(entry => entry.payload.value.odbAddress.indexOf(nameToSpaceName(this._name)) !== -1)
+      if (!entry) {
+        await this._rootStore.add({ type: 'space', DID: this.DID, odbAddress: spaceAddress })
```

Comment:
👍

Reply:
then in future, rootstore <-> space/did link can be verified by resolving the did now? right?

Reply:
Precisely :)